File: Contests/2019-2020/Bronze/Feb/breedflip.py
import sys
sys.stdin = open("breedflip.in", "r")
sys.stdout = open("breedflip.out", "w")

n = int(input())
a = input()
b = input()
flag = 1
cnt = 0
for i in range(n):
    if a[i] != b[i] and flag:
        cnt += 1
        flag = 0
    elif a[i] == b[i]:
        flag = 1
print(cnt)

File: Contests/2019-2020/Bronze/Feb/breedflip.cpp
#include <bits/stdc++.h>
#pragma GCC optimize(2)
using namespace std;
typedef long long ll;
const int MAXN = 105;

int n, cnt, flag = 1;
string a, b;
int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    freopen("breedflip.in", "r", stdin);
    freopen("breedflip.out", "w", stdout);
    cin >> n >> a >> b;
    for (int i = 0; i < n; ++i) {
        if ((a[i]!=b[i]) && flag) {
            ++cnt;
            flag = 0;
        }
        if (a[i]==b[i]) flag = 1;
    }
    cout << cnt << endl;
    return 0;
}